controllers/clusterresources/postgresqluser_controller.go

```diff
@@ -259,12 +321,72 @@ func (r *PostgreSQLUserReconciler) createUser(
 	createUserQuery := fmt.Sprintf(`CREATE USER "%s" WITH PASSWORD '%s'`, newUserName, newPassword)
 	_, err = conn.Exec(ctx, createUserQuery)
 	if err != nil {
-		return fmt.Errorf("cannot execute creation user query in postgresql, err: %w", err)
+		return fmt.Errorf("cannot execute creation user query in PostgreSQL, err: %w", err)
 	}
 
 	return nil
 }
 
+func (r *PostgreSQLUserReconciler) deleteUser(
+	ctx context.Context,
+	newUserName string,
+	defaultUserSecretNamespacedName clusterresourcesv1beta1.NamespacedName,
+) error {
+	defaultUsername, defaultPassword, clusterName, err := r.getDefaultPostgreSQLUserCreds(ctx, defaultUserSecretNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	serviceName := fmt.Sprintf(models.ExposeServiceNameTemplate, clusterName)
+	host := fmt.Sprintf("%s.%s", serviceName, defaultUserSecretNamespacedName.Namespace)
+
+	dbURL := fmt.Sprintf("postgres://%s:%s@%s:%d/%s?target_session_attrs=read-write",
+		defaultUsername, defaultPassword, host, models.DefaultPgDbPortValue, models.DefaultPgDbNameValue)
+
+	conn, err := pgx.Connect(ctx, dbURL)
+	if err != nil {
+		return fmt.Errorf("cannot establish a connection with a PostgreSQL server, err: %w", err)
+	}
+	defer conn.Close(ctx)
+
+	createUserQuery := fmt.Sprintf(`DROP USER IF EXISTS "%s"`, newUserName)
+	_, err = conn.Exec(ctx, createUserQuery)
+	if err != nil {
+		return fmt.Errorf("cannot execute creation user query in PostgreSQL, err: %w", err)
+	}
+
+	return nil
+}
+
+func (r *PostgreSQLUserReconciler) getDefaultPostgreSQLUserCreds(
+	ctx context.Context,
+	defaultUserSecretNamespacedName clusterresourcesv1beta1.NamespacedName,
+) (string, string, string, error) {
+	defaultUserSecret := &k8sCore.Secret{}
+
+	namespacedName := types.NamespacedName{
+		Namespace: defaultUserSecretNamespacedName.Namespace,
+		Name:      defaultUserSecretNamespacedName.Name,
+	}
+	err := r.Get(ctx, namespacedName, defaultUserSecret)
+	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			return "", "", "", fmt.Errorf("cannot get default PostgreSQL user secret, user reference: %v, err: %w", defaultUserSecretNamespacedName, err)
+		}
+
+		return "", "", "", err
+	}
+
+	defaultUsername, defaultPassword, err := getUserCreds(defaultUserSecret)
+	if err != nil {
+		return "", "", "", fmt.Errorf("cannot get default PostgreSQL user credentials, user reference: %v, err: %w", defaultUserSecretNamespacedName, err)
+	}
+
+	clusterName := defaultUserSecret.Labels[models.ControlledByLabel]
+
+	return defaultUsername, defaultPassword, clusterName, nil
```

Comment:
Maybe it would be better to return structure end error instead of multiple arguments

Reply:
Fixed